File: nodejs/tests/connection/spi_test.js
'use strict';

const { SPIConnection } = require('../../packages/periph/src/connection/spi');

const SPI_BUS    = parseInt(process.env.SPI_BUS    || '0', 10);
const SPI_DEVICE = parseInt(process.env.SPI_DEVICE || '0', 10);
const SPI_SPEED  = parseInt(process.env.SPI_SPEED  || '1000000', 10);

let passed = 0;
let failed = 0;

function checkTrue(label, condition) {
    if (condition) { console.log('PASS', label); passed++; }
    else           { console.log('FAIL', label); failed++; }
}

async function main() {
    const connection = new SPIConnection(SPI_BUS, SPI_DEVICE, { maxSpeedHz: SPI_SPEED });

    await connection.write(Buffer.from([0x00]));
    checkTrue('write accepted', true);

    const readBuf = await connection.read(1);
    checkTrue('read returns 1 byte', readBuf.length === 1);

    const wrBuf = await connection.writeRead(Buffer.from([0x00]), 1);
    checkTrue('writeRead returns 1 byte', wrBuf.length === 1);

    await connection.close();

    console.log(`===DONE: ${passed} passed, ${failed} failed===`);
    process.exit(failed === 0 ? 0 : 1);
}

main();

File: python/tests/connection/spi_test.py
import _testconfig as cfg
from machine import SPI, Pin
from periph.connection.spi_micropython import SPIConnection

passed = 0
failed = 0


def check_true(label, condition):
    global passed, failed
    if condition:
        print('PASS', label)
        passed += 1
    else:
        print('FAIL', label)
        failed += 1


spi = SPI(cfg.SPI_ID, baudrate=cfg.FREQ, sck=Pin(cfg.SCK), mosi=Pin(cfg.MOSI), miso=Pin(cfg.MISO))
cs  = Pin(cfg.CS, Pin.OUT, value=1)
connection = SPIConnection(spi, cs)

connection.write(bytes([0x00]))
check_true('write accepted', True)

data = connection.read(1)
check_true('read returns 1 byte', len(data) == 1)

data = connection.write_read(bytes([0x00]), 1)
check_true('write_read returns 1 byte', len(data) == 1)

print('===DONE: {} passed, {} failed==='.format(passed, failed))
